services/controller/src/plz/controller/instances/docker.py

```diff
@@ -112,27 +112,30 @@ def container_state(self) -> Optional[ContainerState]:
 
     def release(self,
                 results_storage: ResultsStorage,
-                _: int,
+                idle_since_timestamp: int,
                 release_container: bool = True):
         if not release_container:
             # Everything to release here is about the container
             return
         with self._lock:
             self.stop_execution()
-            self._publish_results(results_storage)
+            self._publish_results(results_storage,
+                                  finish_timestamp=idle_since_timestamp)
             # Check that we could collect the logs before destroying the
             # container
             if not results_storage.is_finished(self.execution_id):
                 raise CouldNotGetOutputException(
                     f'Couldn\'t read the results for {self.execution_id}')
             self._cleanup()
 
-    def _publish_results(self, results_storage: ResultsStorage):
+    def _publish_results(self, results_storage: ResultsStorage,
+                         finish_timestamp: int):
         results_storage.publish(
             self.get_execution_id(),
             exit_status=self.exit_status(),
             logs=self.logs(since=None),
-            output_tarball=self.output_files_tarball())
+            output_tarball=self.output_files_tarball(),
+            finish_timestamp=finish_timestamp)
```

Comment:
Are we recording the starting timestamp too?

Reply:
Nope. I wanted to go for "smallest working thing". I'd need to store the starting timestamp in a tag too, for the same reason the finish timestamp is. This pr is big enough I think.

Reply:
I have put the finish timestamp as to have a minimal "metadata = start_metadata + finish_metadata" working. The final metadata will include "results" as well, outputted by the job running

Reply:
Cool.